backend/kernelCI_app/views/treeDetailsSlowView.py

```diff
@@ -1,49 +1,82 @@
 from django.http import JsonResponse, HttpResponseBadRequest
 from django.views import View
-from kernelCI_app.utils import (FilterParams, extract_error_message,
-                                extract_platform, InvalidComparisonOP, getErrorResponseBody, toIntOrDefault)
+from kernelCI_app.utils import (
+    FilterParams,
+    extract_error_message,
+    extract_platform,
+    InvalidComparisonOP,
+    getErrorResponseBody,
+    toIntOrDefault,
+)
 from django.db import connection
 
 
 class TreeDetailsSlow(View):
-    def __processFilters(self, request):
-        filterTestStatus = ["FAIL", "MISS", "PASS", "DONE", "ERROR", "SKIP"]
+    def __processFilters(self, request):  # noqa: max-complexity=13
+        filterTestStatus = set()
         filterTestDurationMin, filterTestDurationMax = None, None
-        filterBootStatus = ["FAIL", "MISS", "PASS", "DONE", "ERROR", "SKIP"]
+        filterBootStatus = set()
         filterBootDurationMin, filterBootDurationMax = None, None
+        filterTreeDetailsConfigs = set()
+        filterTreeDetailsCompiler = set()
+        filterArchitecture = set()
 
         try:
             filter_params = FilterParams(request)
-            for f in filter_params.filters:
-                field = f["field"]
-                value = f["value"]
-                operation = f["comparison_op"]
+            for current_filter in filter_params.filters:
+                field = current_filter["field"]
+                value = current_filter["value"]
+                operation = current_filter["comparison_op"]
                 if field == "boot.status":
-                    filterBootStatus = value
+                    filterBootStatus.add(value)
                 elif field == "boot.duration":
                     if operation == "lte":
                         filterBootDurationMax = toIntOrDefault(value, None)
                     else:
                         filterBootDurationMin = toIntOrDefault(value, None)
                 if field == "test.status":
-                    filterTestStatus = value
+                    filterTestStatus.add(value)
+                if field == "config_name":
```

Comment:
Have you considered using a `switch/case`? I'm not sure, but maybe a `switch/case` would also help us not need the `# noqa: max-complexity=13`

Reply:
Heads up: switch/case is match in python.

The way to reduce complexity is to divide this function in smaller ones

Reply:
it wasn't needed just splitted into functions